sign/sign.go

```diff
@@ -124,7 +124,41 @@ func (s *Signer) SignImage(reference string) (*SignedObject, error) {
 func (s *Signer) SignFile(path string) (*SignedObject, error) {
 	s.log.Infof("Signing file path: %s", path)
 
-	// TODO: unimplemented
+	resetFn, err := s.enableExperimental()
+	if err != nil {
+		return nil, err
+	}
+	defer resetFn()
+
+	ko := sign.KeyOpts{
+		KeyRef:     s.options.PrivateKeyPath,
+		PassFunc:   s.options.PassFunc,
+		FulcioURL:  cliOpts.DefaultFulcioURL,
+		RekorURL:   cliOpts.DefaultRekorURL,
+		OIDCIssuer: cliOpts.DefaultOIDCIssuerURL,
+
+		InsecureSkipFulcioVerify: false,
+	}
+
+	regOpts := cliOpts.RegistryOptions{
+		AllowInsecure: s.options.AllowInsecure,
+	}
+
+	outputSignature := ""
+	if s.options.OutputSignaturePath != "" {
+		outputSignature = s.options.OutputSignaturePath
+	}
+
+	outputCertificate := ""
+	if s.options.OutputCertificatePath != "" {
+		outputCertificate = s.options.OutputCertificatePath
+	}
+
+	if err := s.impl.SignFileInternal(context.Background(), ko, regOpts, path,
```

Comment:
Let’s use a context with timeout here.

Reply:
I kept context and timeout both to keep the API consistent. Looking back, I think keeping both is creating confusion. So, I have removed passed timeout and instead passing context with a cancellation. Internally, cosign uses this context if timeout is set to 0.

Reply:
Yeah that's confusing. Following up in cosign about that one: https://github.com/sigstore/cosign/pull/1454